sdk/keyvault/Azure.Security.KeyVault.Secrets/tests/SecretsTests.cs

```diff
@@ -0,0 +1,259 @@
+            var client = new SecretClient(VaultUri, AzureCredential.Default);
+
+            Secret setResult = await client.SetAsync("CrudBasic", "CrudBasicValue1");
+
+            Secret getResult = await client.GetAsync("CrudBasic");
+
+            Assert.AreEqual("CrudBasic", setResult.Name);
+            Assert.AreEqual(VaultUri, setResult.Vault);
+
+            AssertSecretsEqual(setResult, getResult);
+
+            getResult.Enabled = false;
+            SecretBase updateResult = await client.UpdateAsync(getResult);
+
+            AssertSecretsEqual(getResult, updateResult);
+
+            DeletedSecret deleteResult = await client.DeleteAsync("CrudBasic");
+
+            AssertSecretsEqual(updateResult, deleteResult);
+        }
+
+        [Test]
+        public async Task CrudWithExtendedProps()
+        {
+            var client = new SecretClient(VaultUri, AzureCredential.Default);
+
+            var secret = new Secret("CrudWithExtendedProps", "CrudWithExtendedPropsValue1")
+            {
+                ContentType = "password",
+                NotBefore = UtcNowMs() + TimeSpan.FromDays(1),
+                Expires = UtcNowMs() + TimeSpan.FromDays(90)
+            };
+
+            Secret setResult = await client.SetAsync(secret);
+
+            Assert.AreEqual("password", setResult.ContentType);
+
+            Secret getResult = await client.GetAsync("CrudWithExtendedProps");
+
+            AssertSecretsEqual(setResult, getResult);
+
+            DeletedSecret deleteResult = await client.DeleteAsync("CrudWithExtendedProps");
+
+            // remove the value which is not set on the deleted response
+            typeof(Secret).GetProperty(nameof(setResult.Value)).SetValue(setResult, null);
+
+            AssertSecretsEqual(setResult, deleteResult);
+        }
+
+        [Test]
+        public async Task BackupRestore()
+        {
+            var backupPath = Path.GetTempFileName();
+
+            try
+            {
+                var client = new SecretClient(VaultUri, AzureCredential.Default);
+
+                Secret setResult = await client.SetAsync("BackupRestore", "BackupRestore");
+
+                byte[] backup = await client.BackupAsync("BackupRestore");
+
+                await client.DeleteAsync("BackupRestore");
+
+                Secret restoreResult = await client.RestoreAsync(backup);
+
+                // remove the vaule which is not set in the restore response
+                typeof(Secret).GetProperty(nameof(setResult.Value)).SetValue(setResult, null);
+
+                AssertSecretsEqual(setResult, restoreResult);
+            }
+            finally
+            {
+                File.Delete(backupPath);
+            }
+        }
+
+        private DateTime UtcNowMs()
+        {
+            return DateTime.MinValue.ToUniversalTime() + TimeSpan.FromMilliseconds(new TimeSpan(DateTime.UtcNow.Ticks).TotalMilliseconds);
+        }
+
+    }
+
+    public class SecretListTests : KeyVaultTestBase, IDisposable
```

Comment:
Why do we need a separate class for these tests? Arent the `SecretTests ` also live ?